src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -69,9 +69,7 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
             this.programmingExercise = programmingExercise;
             this.isExamExercise = !!this.programmingExercise.exerciseGroup;
             this.courseId = this.isExamExercise ? this.programmingExercise.exerciseGroup!.exam!.course!.id! : this.programmingExercise.course!.id!;
-            this.programmingExercise.isAtLeastTutor = this.accountService.isAtLeastTutorForExercise(this.programmingExercise);
-            this.programmingExercise.isAtLeastEditor = this.accountService.isAtLeastEditorForExercise(this.programmingExercise);
-            this.programmingExercise.isAtLeastInstructor = this.accountService.isAtLeastInstructorForExercise(this.programmingExercise);
+            this.accountService.setAccessRightsForExercise(this.programmingExercise);
```

Comment:
In the future this method call should move into `courseService` when the response is handled.

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3742/commits/1fe9a87043e8262f95c34eacc86074bf52f23523